### ISSUE TITLE
Library needs to be more strict with missing dictionaries

### ISSUE THREAD

**Issue by depryf** [imsweb/naaccr-xml#196]
When I designed the library, I didn't want to crash the reading process because of a missing dictionary; I wanted a calling software to be able to report to the user "hey, the dictionary is missing, do you want to go on and just ignore unknown item?".

And that works very will in some software, but other software/small programs are struggling with this. It is technically not correct to try to read an XML file without providing the dictionaries that are needed. And it is confusing that the library doesn't reports a missing dictionary by default. 

I would like to change the default behavior to throw an exception if a reading is initialized for a file but a dictionary is missing.

I will tie that to an option so that some software can keep the current behavior if they prefer...

**Comment by depryf** [comment]
@howew I am notifying you on this issue because it is relevant to a recent conversation we had.

I added this issue a while ago, and so it's not a result of that conversation.

But I am planning on changing the behavior of the library to throw an IOException if not all the referenced dictionaries are provided to the reader. 

Some applications properly handle missing dictionaries, and so there will be a new option to not throw the exception (which is the current behavior). But I feel like reporting an error is what most application would want and so I am going to make that the default behavior.

You might want to keep an eye on this since it might affect NAACCR*Prep.

**Comment by depryf** [comment]
This is done, new option is called "allowMissingDictionaries". 

I initially said I would throw an exception by default, but I changed my mind. I got scarred of the side affects of changing the default behavior.

Since the library doesn't throw an exception right now, the default behavior of that new option is to allow missing dictionaries.

Programs that want to ensure they always provide the dictionaries should set the new option to false.
